src/server/engine_shard.cc

```diff
@@ -59,6 +59,8 @@ ABSL_FLAG(double, eviction_memory_budget_threshold, 0.1,
           "Eviction starts when the free memory (including RSS memory) drops below "
           "eviction_memory_budget_threshold * max_memory_limit.");
 
+ABSL_FLAG(bool, background_heartbeat, false, "Whether to run heartbeat as a background fiber");
```

Comment:
why do you think we need to convert heartbeat into a background fiber?
I thought we convert  the snapshotting fiber first, see how it actually behaves before starting to expand it to more usecases. Moreover, I would not pick this fiber as my second candidate.